Tests/youdaTests/IAPServiceMockTest.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  youda
+//
+//  Copyright (c) 2020 Industrial Binaries
+//  MIT license, see LICENSE file for details
+//
+
+import StoreKit
+import XCTest
+@testable import youda
+
+final class IAPServiceMockTest: XCTestCase {
+  func testAvailableProducts() {
+    let apiService = IAPServiceMock(
+      availableProducts: .mockAvailableProducts,
+      purchasedProducts: []
+    )
+
+    XCTAssertEqual(apiService.availableProducts.first?.productIdentifier, .purchasedAvailableIdentifier)
+  }
+
+  func testPurchasedProducts() {
+    let apiService = IAPServiceMock(
+      availableProducts: [],
+      purchasedProducts: .mockPurchasedProducts
+    )
+
+    XCTAssertEqual(apiService.purchasedProducts.first?.productIdentifier, .purchasedProductIdentifier)
+  }
+
+  func testBuyProductNotification() {
+    let apiService = IAPServiceMock(
+      availableProducts: .mockAvailableProducts,
+      purchasedProducts: []
+    )
+
+    let expectation = XCTestExpectation(description: "Buy products timeout")
+    NotificationCenter.default.addObserver(
+      forName: .subscriptionChange,
+      object: nil,
+      queue: nil
+    ) { _ in
+      expectation.fulfill()
+    }
+
+    apiService.buy(product: .purchasedAvailableIdentifier)
+
+    wait(for: [expectation], timeout: 5)
+    XCTAssertEqual(apiService.purchasedProducts.first?.productIdentifier, .purchasedAvailableIdentifier)
+  }
+
+  func testRestoreProductsNotification() {
+    let apiService = IAPServiceMock(
+      availableProducts: .mockAvailableProducts,
+      purchasedProducts: []
+    )
+
+    let expectation = XCTestExpectation(description: "Restore products timeout")
+    NotificationCenter.default.addObserver(
+      forName: .subscriptionChange,
+      object: nil,
+      queue: nil
+    ) { _ in
+      expectation.fulfill()
+    }
+
+    apiService.restoreProducts()
+
+    wait(for: [expectation], timeout: 5)
+    XCTAssertEqual(apiService.purchasedProducts.first?.productIdentifier, .purchasedAvailableIdentifier)
+  }
+
+  private var expectation: XCTestExpectation!
```

Comment:
I see what you're doing here, but it feels a little bit strange. What if you need 2 expectations? Let's talk IRL